dashboard/app/models/user.rb

```diff
@@ -1878,6 +1878,34 @@ def recent_pl_courses_and_units(exclude_primary_script)
     user_course_data + user_script_data
   end
 
+  def pl_units_started
```

Comment:
I based a lot of this logic on `recent_pl_courses_and_units` [above](https://github.com/code-dot-org/code-dot-org/pull/56119/files#diff-0a5e79b422e8d2a2b8e5d9cc2203f62899ee74b65da749306b3d5f7c8b21a2d2R1848), but we needed a slightly different format and only units in this case (see [relevant discussion](https://www.figma.com/file/sZ6d1uhJWFksJ6j3Fnmbnc?node-id=1399:6334&mode=dev#679809654))